spec/std/syscall_spec.cr

```diff
@@ -0,0 +1,31 @@
+{% skip_file unless flag?(:linux) %}
+
+require "spec"
+require "syscall"
+
+private module TestSyscall
```

Comment:
This private namespace shouldn't be necessary (same for random). `Syscall.def_syscall` should just be a public macro and callable from anywhere. Then the following defs can be `private Syscall.def_syscall xxx` in the file scope.

Reply:
Done.